03/Product.js

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+
+export default class Product extends React.Component {
+	render() {
+		const { item, onClick, isCategory, isCart } = this.props;
+
+		if (isCategory) {
+			return (
+				<li>
+					{item.name} Cena: {item.price}zł
+					<button
+						disabled={false}
+						onClick={onClick}>
+						Dodaj do koszyka
+					</button>
+				</li>
+			);
+		}
+
+		if (isCart) {
+			return (
+				<li>
+					{item.name} Cena: {item.price}zł
+					<button
+						disabled={false}
+						onClick={onClick}>
+						Usuń z koszyka
+					</button>
+				</li>
+			);
+		}
+	}
```

Comment:
Zwróć uwagę, że mamy tutaj mnóstwo powielonego kodu, jedyne co się różni to napis i to ten element można warunkować np. :)
```
<button
  disabled={false}
  onClick={onClick}>
    {isCategory ? 'dodaj' : 'usuń'}
 </button>
```